search/search_repository.go

```diff
@@ -307,6 +314,24 @@ func parseMap(queryMap map[string]interface{}, q *Query) {
 	}
 }
 
+func parseOptions(queryMap map[string]interface{}, q *Query) {
```

Comment:
The way I see this function is that it only deals with extracting the query portion of from the map. So why not just return a `QueryOptions` struct pointer (or `nil`) and remove the `Query` parameter. There's no recursion here, so need to make things more complicated than needed be. On the caller side this simplifies things as well to `q.Options = parseOptions(fm)`.

Reply:
Done. f783e2b322961c038873d51c0866bb7d53096dd4